frontend/src/metabase/dashboard/components/PublicLinkPopover/PublicLinkPopover.unit.spec.tsx

```diff
@@ -0,0 +1,239 @@
+      const { createPublicLink } = setup({ hasUUID: false });
+
+      expect(createPublicLink).toHaveBeenCalledTimes(1);
+    });
+
+    it("should not call createPublicLink when isOpen is false", () => {
+      const { createPublicLink } = setup({ isOpen: false, hasUUID: false });
+
+      expect(createPublicLink).not.toHaveBeenCalled();
+    });
+
+    it("should call createPublicLink when uuid is null and the popover is opened", async () => {
+      const { createPublicLink } = setup({ hasUUID: false, isOpen: false });
+
+      userEvent.click(screen.getByTestId("target"));
+
+      expect(await screen.findByText("sample-public-link")).toBeInTheDocument();
+      expect(createPublicLink).toHaveBeenCalledTimes(1);
+    });
+
+    it("should not call createPublicLink when uuid is not null and the popover is opened", async () => {
+      const { createPublicLink } = setup({ hasUUID: true, isOpen: false });
+
+      userEvent.click(screen.getByTestId("target"));
+
+      expect(createPublicLink).not.toHaveBeenCalled();
+      expect(await screen.findByText("sample-public-link")).toBeInTheDocument();
+    });
+  });
+
+  describe("when deleting public links", () => {
+    it("should call deletePublicLink and onClose when `Remove this public link` is clicked", () => {
+      const { deletePublicLink, onClose } = setup({
+        hasUUID: true,
+        isOpen: true,
+        isAdmin: true,
+      });
+
+      userEvent.click(screen.getByText("Remove this public link"));
+
+      expect(deletePublicLink).toHaveBeenCalledTimes(1);
+      expect(onClose).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe("when copying the link", () => {
+    it("should allow admins to copy the link to the clipboard", async () => {
+      setup({ hasUUID: true, isOpen: true });
+
+      expect(await screen.findByText("sample-public-link")).toBeInTheDocument();
+
+      userEvent.click(screen.getByLabelText("copy icon"));
+
+      expect(await screen.findByText("Copied!")).toBeInTheDocument();
+    });
+
+    it("should allow non-admins to copy the link to the clipboard", async () => {
+      setup({ hasUUID: true, isOpen: true, isAdmin: false });
+
+      expect(await screen.findByText("sample-public-link")).toBeInTheDocument();
+
+      userEvent.click(screen.getByLabelText("copy icon"));
+
+      expect(await screen.findByText("Copied!")).toBeInTheDocument();
+    });
+  });
+
+  describe("when appending extensions to the link", () => {
+    it("should append the extension to the link when the extension is clicked on", async () => {
+      setup({ hasUUID: true, isOpen: true, extensions: ["csv"] });
+
+      expect(await screen.findByText("sample-public-link")).toBeInTheDocument();
+
+      userEvent.click(screen.getByText("csv"));
+      expect(
+        await screen.findByText("sample-public-link.csv"),
+      ).toBeInTheDocument();
+    });
+
+    it("should remove the extension when the extension is clicked on again", async () => {
```

Comment:
This reaffirms my point made in [this comment](https://github.com/metabase/metabase/pull/36107#discussion_r1411789174). I think toggling the extension looks very unintuitive imo.

Reply:
Yep, I totally agree. It feels like it was just shoehorned in